src/VisualStudio/Core/Def/Interactive/VsInteractiveWindowPackage.cs

```diff
@@ -51,15 +51,13 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
 
             // Set both handlers to non-fatal Watson. Never fail-fast the VS process.
             // Any exception that is not recovered from shall be propagated.
-            var nonFatalHandler = new Action<Exception>(static (exception) => WatsonReporter.ReportNonFatal(exception, forceDump: false));
-            InteractiveHostFatalError.Handler = nonFatalHandler;
-            InteractiveHostFatalError.NonFatalHandler = WatsonReporter.ReportNonFatal;
-
-            // Load the Roslyn package so that its FatalError handlers are hooked up.
-            shell.LoadPackage(Guids.RoslynPackageId, out var roslynPackage);
+            FaultReporter.InitializeFatalErrorHandlers();
+            FatalError.CopyHandlerTo(typeof(InteractiveHostFatalError).Assembly);
 
             // Explicitly set up FatalError handlers for the InteractiveWindowPackage.
-            var fatalHandler = nonFatalHandler;
+            Action<Exception> fatalHandler = e => FaultReporter.ReportFault(e, VisualStudio.Telemetry.FaultSeverity.Critical, forceDump: false);
+            Action<Exception> nonFatalHandler = e => FaultReporter.ReportFault(e, VisualStudio.Telemetry.FaultSeverity.General, forceDump: false);
+
             SetErrorHandlers(typeof(IInteractiveWindow).Assembly, fatalHandler, nonFatalHandler);
             SetErrorHandlers(typeof(IVsInteractiveWindow).Assembly, fatalHandler, nonFatalHandler);
```

Comment:
There is a separate piece of work (outside the scope of this PR) to remove the reflection against the interactive window package. The interactive window package should just be doing it's own thing directly.